en/starter/generator/docker_container_tutorial.md

```diff
@@ -1,148 +1,154 @@
+    ? Select capabilities: (Press <space> to select, <a> to toggle all, <i> to inverse selection)
+    ❯ ◯ Static web file serving
+      ◯ Swagger UI
+      ◯ Embedded metrics dashboard
+      ◉ Docker files
+
+<span class="arrow">&#8227;</span> Press **Enter** to accept the default of not generating code from a [swagger](core_concepts.html#endpoints-from-swagger-file) specification in the scaffolding.
+
+    ? Select endpoints to generate: (Press <space> to select, <a> to toggle all, <i> to inverse selection)
+    ❯ ◯ Swagger file serving endpoint
+      ◯ Endpoints from a swagger file
+
+<span class="arrow">&#8227;</span> Press **Enter** to accept the default of not generating a Swift server SDK from a swagger file in the scaffolding.
+
+    ? Would you like to generate a Swift server SDK from a Swagger file? (y/N)
+
+<span class="arrow">&#8227;</span> Press **Enter** to accept the default of not including any boilerplate for [services](core_concepts.html#services) in the scaffolding.
+
+    ? Generate boilerplate for services: (Press <space> to select, <a> to toggle all, <i> to inverse selection)
+    ❯ ◯ Cloudant / CouchDB
+      ◯ Redis
+      ◯ MongoDB
+      ◯ PostgreSQL
+      ◯ Object Storage
+      ◯ AppID
+      ◯ Auto-scaling
+---
+
+The generator will display messages as it scaffolds and builds the application including:
+
+1.  Initializing the project folder structure.
+
+1.  Creating and compiling default Swift files.
+
+1.  Downloading and installing dependent Swift modules (as if you had manually run `swift build`).
+
+<span class="arrow">&#8227;</span> Now you should have a generated application directory similar to this:
+<pre>
+docker-getting-started/
+├── Sources/
+│   ├── Application
+│   │   ├── Application.swift
+│   │   └── Routes/
+│   └── docker-getting-started
+│       └── main.swift
+├── Package.swift
+├── README.md
+├── config.json
+├── Dockerfile
+├── Dockerfile-tools
+├── .dockerignore
+└── ...
+</pre>
+
+> ![info] For a description of the generated files, take a look at the structure of a generated project in the [Project layout reference](project_layout_reference.html).
+
+<span class="arrow">&#8227;</span> Change to the application directory:
+
+    $ cd docker-getting-started
+
+<span class="arrow">&#8227;</span> Build the Docker run and Docker build tools images:
+
+```bash
+$ docker build -t myapp-run .
+$ docker build -t myapp-build -f Dockerfile-tools .
+```
+
+You may customize the names of these images by specifying a different value after the `-t` option.
+
+<span class="arrow">&#8227;</span> To compile the application using the build tools Docker image, run:
+
+```bash
+$ docker run -v $PWD:/root/project -w /root/project myapp-build /root/utils/tools-utils.sh build release
+```
+
+<span class="arrow">&#8227;</span> To run the application in the Docker container:
+
+```bash
+$ docker run -it -p 8080:8080 -v $PWD:/root/project -w /root/project myapp-run sh -c .build-ubuntu/release/docker-getting-started
```

Comment:
For all the above docker commands: have we tested these? I know there were some changes made since last release that affect these commands but I don't know if they were already updated to take them into account.

Reply:
This would also affect the doc in core_concepts.md

Reply:
In a PR